core/src/main/resources/reference.conf

```diff
@@ -655,3 +655,11 @@ workflow-state-callback {
   # max-retries = 10
 
 }
+
+# Config for running private workflows. Currently only supported in Azure
+private-workflows {
+  enabled = false
+  auth {
+    azure = false
+  }
+}
```

Comment:
+1 to combining this with the token access config.

Reply:
I had added this here as there wasn't an `enabled` config value in `GithubAuthVending` initially. But now I can combine it. Do we want the enabling of  `GithubAuthVending` to be only for Azure environment? Or would the value for `GithubAuthVending.confog.enabled` be always false but overriden in Azure helm charts in future?

Reply:
Actually we might need some way of figuring out which environment Cromwell is running in or not because we can only do [AzureCredentials.getToken() here](https://github.com/broadinstitute/cromwell/pull/7373/files#diff-6ce4458da3c6a4dfc783abb882caea399405deb4401049bc4f3c83874b395fb1R197) only if Cromwell is running on Azure

Reply:
Maybe we can incorporate the `auth.azure` config key into the other config, and always fall back to it if there isn't a token attached to the incoming request? 

In practice we need two different auth strategies to support what we're trying to do on Azure, but that's due to the specific way we've set up these multiple instances. It would be nice to hide that difference from the configurer, and them think of it in terms of: Cromwell will use a token attached to an incoming request when it can, and will otherwise get a token from <insert configured auth strategy>.

Reply:
I can move `auth.azure` inside `GithubAuthVending.config` 👍 
>Cromwell will use a token attached to an incoming request when it can

For workflow submissions, if we extract tokens from the request, we would need to store that token in the database or in some form in memory no? Because when workflow is being materialized it doesn't have access to the original request and its headers as the workflow upon submission is store in WorkflowStore table? If my understanding is right then in this case we would always need to get the token using `AzureCredentials.getToken()`

Reply:
Oh, by "when it can" I mean in the parts of the process where that token is already available :) Not suggesting we make it available in places it isn't already.